### PR TITLE
core: Adding kwargs to cache to be able to add metadat

### PR DESCRIPTION
**Pull request by romainrey**
- **Description:**
Adding the possibility to pass kwargs to the cache functions. 
[Discussed here](https://github.com/langchain-ai/langchain/discussions/18215)

### CODE REVIEW

**Review by hwchase17 — COMMENTED** [review]
whats a use case?

**Comment by hwchase17** [comment]
nvr mind i see the discussion. thoughts @eyurtsev ?

**Comment by romainrey** [comment]
I think we might need to adjust the pr slightly if that is something you guys want to push.
Happy to do it if you are ok on the idea/

File: libs/core/langchain_core/caches.py [hunk 1]
```diff
@@ -34,23 +34,23 @@ class BaseCache(ABC):
     """Base interface for cache."""
 
     @abstractmethod
-    def lookup(self, prompt: str, llm_string: str) -> Optional[RETURN_VAL_TYPE]:
+    def lookup(self, prompt: str, llm_string: str, cache_params: Any) -> Optional[RETURN_VAL_TYPE]:
```
Comment (eyurtsev) [inline libs/core/langchain_core/caches.py:37]: A few questions:

* What is the type of cache_params? Do you imagine a flat dict would be sufficient?
* What are the semantics for cache_params when doing look up?
* What are the semantics when doing an update?

**Review by eyurtsev — COMMENTED** [review]
The use case makes sense. Would love to figure out the cleanest way to accommodate. Left a question on the PR -- could you tell me what you imagine for the type and the semantics for implementors of cache?

**Comment by eyurtsev** [comment]
Closing as stale

**Pull request closed without merging**